chips/stm32f4xx/src/flash.rs

```diff
@@ -0,0 +1,711 @@
+        } else if frequency_mhz <= 150 {
+            FlashLatency::Latency4
+        } else {
+            FlashLatency::Latency5
+        }
+    }
+
+    #[cfg(any(feature = "stm32f410", feature = "stm32f411", feature = "stm32f412"))]
+    fn get_number_wait_cycles_based_on_frequency(&self, frequency_mhz: usize) -> FlashLatency {
+        if frequency_mhz <= 30 {
+            FlashLatency::Latency0
+        } else if frequency_mhz <= 64 {
+            FlashLatency::Latency1
+        } else if frequency_mhz <= 90 {
+            FlashLatency::Latency2
+        } else {
+            FlashLatency::Latency3
+        }
+    }
+
+    #[cfg(any(feature = "stm32f413", feature = "stm32f423"))]
+    fn get_number_wait_cycles_based_on_frequency(&self, frequency_mhz: usize) -> FlashLatency {
+        if frequency_mhz <= 25 {
+            FlashLatency::Latency0
+        } else if frequency_mhz <= 50 {
+            FlashLatency::Latency1
+        } else if frequency_mhz <= 75 {
+            FlashLatency::Latency2
+        } else {
+            FlashLatency::Latency3
+        }
+    }
+
+    // Return the current flash latency
+    pub(crate) fn get_latency(&self) -> FlashLatency {
+        #[cfg(not(any(
+            feature = "stm32f405",
+            feature = "stm32f415",
+            feature = "stm32f407",
+            feature = "stm32f417"
+        )))]
+        match self.registers.acr.read(ACR::LATENCY) {
+            0 => FlashLatency::Latency0,
+            1 => FlashLatency::Latency1,
+            2 => FlashLatency::Latency2,
+            3 => FlashLatency::Latency3,
+            4 => FlashLatency::Latency4,
+            5 => FlashLatency::Latency5,
+            6 => FlashLatency::Latency6,
+            7 => FlashLatency::Latency7,
+            8 => FlashLatency::Latency8,
+            9 => FlashLatency::Latency9,
+            10 => FlashLatency::Latency10,
+            11 => FlashLatency::Latency11,
+            12 => FlashLatency::Latency12,
+            13 => FlashLatency::Latency13,
+            14 => FlashLatency::Latency14,
+            // The hardware allows 4-bit latency values
+            _ => FlashLatency::Latency15,
+        }
+
+        #[cfg(any(
+            feature = "stm32f405",
+            feature = "stm32f415",
+            feature = "stm32f407",
+            feature = "stm32f417"
+        ))]
+        match self.registers.acr.read(ACR::LATENCY) {
+            0 => FlashLatency::Latency0,
+            1 => FlashLatency::Latency1,
+            2 => FlashLatency::Latency2,
+            3 => FlashLatency::Latency3,
+            4 => FlashLatency::Latency4,
+            5 => FlashLatency::Latency5,
+            6 => FlashLatency::Latency6,
+            // The hardware allows 3-bit latency values
+            _ => FlashLatency::Latency7,
+        }
+    }
+
+    // TODO: Take into the account the power supply
+    //
+    // NOTE: This method is pub(crate) to prevent a capsule from modifying the flash latency. Flash
```

Comment:
A capsule wouldn't be able to call this function even if it was just `pub` because the capsule crates do not have the stmX crates as dependencies.

Reply:
The comment text was wrong. It's `pub(crate)` to prevent modifying the flash latency from the board main file. Fixed it.